server/pypi/packages/opencv-python/patches/chaquopy.patch

```diff
@@ -76,19 +74,23 @@ index 27874283e..ee76ea661 100644
 +    build_headless = "headless" in os.environ["PKG_NAME"]
 +
      build_java = "ON" if get_build_env_var_by_name("java") else "OFF"
+     build_rolling = get_build_env_var_by_name("rolling")
  
-     if sys.version_info[:2] >= (3, 6):
-@@ -38,12 +41,26 @@
-     numpy_version = "numpy>=%s" % minimum_supported_numpy
+@@ -36,17 +39,26 @@
+     ]
  
      python_version = cmaker.CMaker.get_python_version()
--    python_lib_path = cmaker.CMaker.get_python_library(python_version).replace(
--        "\\", "/"
--    )
+-    python_lib_path = cmaker.CMaker.get_python_library(python_version) or ""
+-    # HACK: For Scikit-build 0.17.3 and newer that returns None or empty sptring for PYTHON_LIBRARY in manylinux2014
+-    # A small release related to PYTHON_LIBRARY handling changes in 0.17.2; scikit-build 0.17.3 returns an empty string from get_python_library if no Python library is present (like on manylinux), where 0.17.2 returned None, and previous versions returned a non-existent path. Note that adding REQUIRED to find_package(PythonLibs will fail, but it is incorrect (you must not link to libPython.so) and was really just injecting a non-existent path before.
```

Comment:
The meta.yaml file specifies `scikit-build 0.14.0`, so is this patch actually necessary?

If it is necessary, it should be marked with the word "Chaquopy" as the other patches are, to make it obvious when looking at the patched source code.

Also, if you've taken this workaround from somewhere else, please include a URL for the source.

Reply:
OK, sorry, I misunderstood the diff of diffs. The "HACK" code is basically a development of the previous assignment to `python_lib_path`, so what you have here is fine.